File: LAB1/caeser.cpp
#include <iostream>
#include <cstring>
using namespace std;
 

string encrypt(string text, int key)
{
    string result = "";
 
    for (int i = 0; i < text.length(); i++) {
        
        if (isupper(text[i]))
            result += char(int(text[i] + key - 65) % 26 + 65);
            
        else if (isspace(text[i]))
          result+=" ";
 

        else
            result += char(int(text[i] + key - 97) % 26 + 97);
    }
 

    return result;
}

string decrypt(string encryption, int key)
{
    string result = "";
 
    for (int i = 1; i < encryption.length()+1; i++) {
        
        if (isupper(encryption[i]))
            result += char(int(encryption[i] - key - 65) % 26 + 65);
            
        else if (isspace(encryption[i]))
          result+=" ";

        else
            result += char(int(encryption[i] - key - 97) % 26 + 97);
    }
 
    return result;
}

 

int main()
{
    string text = "Krisha Pandey";
    int key = 5;
     string encryption =encrypt(text, key);
    cout<<encryption<<endl;
    string decryption=decrypt( encryption,  key);
    cout<<decryption;
}

File: LAB1/caeser.py
def Caesar_encrypt(key,text):
    result=""
    for i in range(len(text)):
        variable=text[i]
        if (variable==" "):
            result+=" "
        if(variable.isupper()):
            result+=chr(((ord(variable)-65+key)%26)+65)
        elif (variable.islower()):
            result+=chr(((ord(variable)-97+key)%26)+97)
    return result

def decryption(encrypt,key):
    result=""
    for i in range(len(encrypt)):
        variable=encrypt[i]
        if (variable==" "):
            result+=" "
        if(variable.isupper()):
            result+=chr(((ord(variable)-65-key)%26)+65)
        elif (variable.islower()):
            result+=chr(((ord(variable)-97-key)%26)+97)
    return result


encrypt=Caesar_encrypt(5,"Krisha Pandey")
decrypt=decryption(encrypt,5)
print(encrypt)
print(decrypt)

 
